src/multidimensional_matrix_operators.jl

```diff
@@ -0,0 +1,191 @@
+end
+
+integrate_boundary(u, D::MultidimensionalMatrixDerivativeOperator, dim) = integrate_boundary(identity, u, D, dim)
+
+weights_boundary(D::MultidimensionalMatrixDerivativeOperator) = get_weight_boundary.(Ref(D), 1:length(grid(D)))
+weights_boundary_scaled(D::MultidimensionalMatrixDerivativeOperator, dim::Int) = get_weight_boundary_scaled.(Ref(D), Ref(dim), 1:length(grid(D)))
+
+"""
+    mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim)
+
+Construct the mass matrix at the boundary of a `MultidimensionalMatrixDerivativeOperator` `D` in direction `dim`.
+The boundary mass matrix is constructed to be mimetic, see
+
+- Jan Glaubitz, Simon-Christian Klein, Jan Nordström, Philipp Öffner (2023)
+  Multi-dimensional summation-by-parts operators for general function spaces: Theory and construction
+  Journal of Computational Physics 491, 112370, [DOI: 10.1016/j.jcp.2023.112370](https://doi.org/10.1016/j.jcp.2023.112370).
+"""
+mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim::Int) = Diagonal(weights_boundary_scaled(D, dim))
+
+function get_weight_boundary(D::MultidimensionalMatrixDerivativeOperator, i::Int)
+    @unpack weights_boundary, on_boundary = D
+    N, _ = size(D)
+    @boundscheck begin
+        @argcheck 1 <= i <= N
+    end
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    j = sum(view(on_boundary, 1:i))
+    @inbounds ω = weights_boundary[j]
+    return ω
+end
+
+function get_weight_boundary_scaled(D::MultidimensionalMatrixDerivativeOperator, dim::Int, i::Int)
+    @unpack normals, on_boundary = D
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    ω = get_weight_boundary(D, i)
+    j = sum(view(on_boundary, 1:i))
+    ω * normals[j][dim]
+end
+
+function Base.show(io::IO, D::MultidimensionalMatrixDerivativeOperator)
+    if get(io, :compact, false)
+        summary(io, D)
+    else
+        x = grid(D)
+        print(io, ndims(D), "-dimensional matrix-based first-derivative operator {T=", eltype(D), "}")
+        print(io, " on ", length(x), " nodes")
+    end
+end
+
+# TODO: mul! How? Depends on direction
+
+"""
+    tensor_product_operator_2D(D)
+
+Create a 2D [`MultidimensionalMatrixDerivativeOperator`](@ref) on a square based on a 1D derivative operator `D` using a tensor product structure.
+
+For the construction, see also:
+- Sigrun Ortleb (2021)
+  Numerical Methods for Fluid Flow:
+    High Order SBP Schemes, IMEX Advection-Diffusion Splitting and Positivity Preservation for Production-Destruction-PDEs
+  Habilitation thesis, University of Kassel, [DOI: 10.17170/kobra-202301037274](https://doi.org/10.17170/kobra-202301037274),
+  Chapter 1.2.4.
+- Magnus Svärd, Jan Nordström (2014)
+  Review of summation-by-parts schemes for initial-boundary-value problems
+  Journal of Computational Physics 268, pp. 17-38, [DOI: 10.1016/j.jcp.2014.02.031](https://doi.org/10.1016/j.jcp.2014.02.031).
+"""
+function tensor_product_operator_2D(D)
```

Comment:
For simplicity, I only allow the same operator in both x- and y-directions. In principle, I think, we can allow two different operators in the two directions, but I would be unsure how to set the boundary weights in the corners as we have two different values in this case. What's your opinion, @ranocha?

Reply:
To underline my point, I just pushed https://github.com/ranocha/SummationByPartsOperators.jl/commit/ba19f39830342555e5785d92fde00a295c13a2ee.

Reply:
This should be resolved after switching to `boundary_indices`.